cmd/tools/gendynamicconfig/main.go

```diff
@@ -0,0 +1,245 @@
+		key:         key,
+		def:         def,
+		description: description,
+	}
+	return s
+}
+
+func New{{.T.Name}}{{.P.Name}}SettingWithConstrainedDefault(key Key, cdef []TypedConstrainedValue[{{.T.GoType}}], description string) {{.T.Name}}{{.P.Name}}Setting {
+	s := {{.T.Name}}{{.P.Name}}Setting{
+		key:         key,
+		cdef:        cdef,
+		description: description,
+	}
+	return s
+}
+
+func (s {{.T.Name}}{{.P.Name}}Setting) Key() Key               { return s.key }
+func (s {{.T.Name}}{{.P.Name}}Setting) Type() Type             { return Type{{.T.Name}} }
+func (s {{.T.Name}}{{.P.Name}}Setting) Precedence() Precedence { return Precedence{{.P.Name}} }
+
+func (s {{.T.Name}}{{.P.Name}}Setting) WithDefault(v {{.T.GoType}}) {{.T.Name}}{{.P.Name}}Setting {
+	newS := s
+	newS.def = v
+	return newS
+}
+
+type {{.T.Name}}PropertyFn{{if .NotGlobal}}With{{.P.Name}}Filter{{end}} func{{.P.GoArgs}} {{.T.GoType}}
+
+func (c *Collection) Get{{.T.Name}}{{if .NotGlobal}}By{{.P.Name}}{{end}}(s {{.T.Name}}{{.P.Name}}Setting) {{.T.Name}}PropertyFn{{if .NotGlobal}}With{{.P.Name}}Filter{{end}} {
+	return func{{.P.GoArgs}} {{.T.GoType}} {
+		return matchAndConvert(
+			c,
+			(Setting[{{.T.GoType}}, func{{.P.GoArgs}}])(s),
+			precedence{{.P.Name}}{{.P.GoArgNames}},
+			convert{{.T.Name}},
+		)
+	}
+}
+
+func Get{{.T.Name}}PropertyFn{{if .NotGlobal}}FilteredBy{{.P.Name}}{{end}}(value {{.T.GoType}}) {{.T.Name}}PropertyFn{{if .NotGlobal}}With{{.P.Name}}Filter{{end}} {
+	return func{{.P.GoArgs}} {{.T.GoType}} {
+		return value
+	}
+}
+`, map[string]any{"T": tp, "P": prec, "NotGlobal": prec.Name != "Global"})
+}
+
+func generate(w io.Writer) {
+	writeTemplatedCode(w, `
+package dynamicconfig
+
+import (
+	"time"
+
+	enumspb "go.temporal.io/api/enums/v1"
+	enumsspb "go.temporal.io/server/api/enums/v1"
+)
+`, nil)
+	for idx, tp := range types {
+		tp.Index = idx
+		generateTypeEnum(w, tp)
+	}
+	for idx, prec := range precedences {
+		prec.Index = idx
+		generatePrecEnum(w, prec)
+	}
+	for _, tp := range types {
+		for _, prec := range precedences {
+			generateType(w, tp, prec)
+		}
+	}
+}
+
+func callWithFile(f func(io.Writer), filename string, licenseText string) {
+	w, err := os.Create(filename + "_gen.go")
+	if err != nil {
+		panic(err)
+	}
+	defer func() {
+		panicIfErr(w.Close())
+	}()
+	if _, err := fmt.Fprintf(w, "%s\n// Code generated by cmd/tools/gendynamicconfig. DO NOT EDIT.\n", licenseText); err != nil {
+		panic(err)
+	}
+	f(w)
+}
+
+func readLicenseFile(path string) string {
+	text, err := os.ReadFile(path)
+	if err != nil {
+		panic(err)
+	}
+	var lines []string
+	for _, line := range strings.Split(string(text), "\n") {
+		lines = append(lines, strings.TrimRight("// "+line, " "))
+	}
+	return strings.Join(lines, "\n") + "\n"
+}
+
+func main() {
+	licenseFlag := flag.String("licence_file", "../../LICENSE", "path to license to copy into header")
+	flag.Parse()
+
+	licenseText := readLicenseFile(*licenseFlag)
+
+	callWithFile(generate, "setting", licenseText)
```

Comment:
Can you call [`format.Source`](https://pkg.go.dev/go/format@go1.22.2#Source) (or Node) to format and check this code before writing it? I'd like to know as soon as possible if this code is invalid

Reply:
I added a call to ParseFile. It's gofmt-clean as generated.